src/app/courses/components/course-card/course-card.component.ts

```diff
@@ -22,7 +26,17 @@ export class CourseCardComponent implements OnInit {
   }
 
   onCourseDelete() {
-    this.deleteCourse.emit(this.course?.id);
+    const dialogRef = this.dialog.open(ModalWindowComponent);
+    this.coursesService.course = this.course
+    // this.deleteCourse.emit(this.course?.id);
   }
 
+  // openDialog() {
+  //   const dialogRef = this.dialog.open(DialogContentExampleDialog);
+
+  //   dialogRef.afterClosed().subscribe(result => {
+  //     console.log(`Dialog result: ${result}`);
+  //   });
+  // }
```

Comment:
Oh, that's cool. 
Could you please implement this way of managing the modal window?

Take a look how to use `MatDialogConfig()` and `dialog.open()` to create reusable dialog with angular material:
https://material.angular.io/components/dialog/overview
https://blog.angular-university.io/angular-material-dialog/

Also, modal window component should be in the core module. I left some suggestions in the comment below.